File: ApiBaseCore/ApiBaseCore.Application/DTOs/AddressDto.cs
using System.Text.Json.Serialization;

namespace ApiBaseCore.Application.DTOs;

public class AddressDto
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Street { get; set; }
    [JsonPropertyName("country_code")]
    public string CountryCode { get; set; }
}

File: ApiBaseCoreJava/src/main/java/com/polancou/apibasecore/application/dtos/AddressDto.java
package com.polancou.apibasecore.application.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class AddressDto {
    private int id;
    private String name;
    private String street;
    
    @JsonProperty("country_code")
    private String countryCode;
}
